CRM/ACL/BAO/ACL.php

```diff
@@ -723,6 +723,40 @@ public static function check($str, $contactID) {
   }
 
   /**
+   * Build a join and where part for a query
+   *
+   * @param int $contactId
+   * @return array - the first key is join part of the query and the second key is the where part of the query
+   */
+  public static function buildAcl($contactId) {
+    $tables = array();
+    $whereTables = array();
+    $whereClause = CRM_ACL_BAO_ACL::whereClause(CRM_Core_Permission::VIEW, $tables, $whereTables, $contactId);
```

Comment:
On further debugging I found out that without any ACL permission it returns 
```$whereClause = ' ( 0 ) ';``` and eventually it gets append with WHERE CLAUSE and eventually no mail will be sent out. Reason why those 10 unit tests were fail. In my opinion introduce an extra parameter like 
```public static function whereClause($type, &$tables, &$whereTables, $contactID = NULL, $strictReturn = FALSE)```
if $strictReturn = TRUE it will return NULL instead of ```( 0 )```